config/app/headers.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+// Default, strict CSP (report only)
+header("Content-Security-Policy-Report-Only: "
+    ."default-src 'self'; "
+    ."style-src 'self'; "
+    ."script-src 'self'; "
+    ."img-src 'self' data:;"
+    ."font-src 'self' data:; "
+    ."connect-src 'self'; "
+    ."frame-src 'self'; "
+    ."form-action 'self'; "
+    ."upgrade-insecure-requests; "
+    ."block-all-mixed-content; "
+    ."report-uri /csp.php");
```

Comment:
Same suggestion as `config/admin/headers.php`:

```suggestion
/**
 * Application: HTTP Content-Security-Policy (CSP) Headers
 */

$directives = [
    "default-src 'self'",
    "connect-src 'self'",
    "font-src 'self' data:",
    "form-action 'self'",
    "frame-src 'self'",
    "img-src 'self' data:",
    "script-src 'self'",
    "style-src 'self'",
    'block-all-mixed-content',
    'upgrade-insecure-requests',
    'report-uri /csp.php',
];

header('Content-Security-Policy-Report-Only: '.implode('; ', $directives));
```

Reply:
I think these HTTP headers should be defined in a service provider (maybe the project's `AppServiceProvider`) to tie into the initial PSR-7 Response object created by Slim.